otherlibs/systhreads/st_stubs.c

```diff
@@ -417,6 +437,10 @@ CAMLprim value caml_thread_initialize(value unit)
   CAMLreturn(Val_unit);
 }
 
+/* Cleanup the thread machinery when the runtime is shut down. Joining the tick
+   thread take 25ms on average / 50ms in the worst case, so we don't do it on
+   program exit. (FIXME) */
```

Comment:
I restored this comment from old systhreads. However, this behaviour does not seem implemented yet in multicore.